src/server/api/general.js

```diff
@@ -44,6 +45,27 @@ export default function useGeneralApi(app) {
     app.use(router.routes());
     const koaBody = koa_body();
 
+    router.get('/state', function *() {
+        const ctx = this;
+        let login_challenge = ctx.session.login_challenge;
+        if (!login_challenge) {
+            login_challenge = secureRandom.randomBuffer(16).toString('hex');
+            ctx.session.login_challenge = login_challenge;
+        }
+        const offchain = {
+            csrf: ctx.csrf,
+            flash: ctx.flash,
+            new_visit: ctx.session.new_visit,
+            account: ctx.session.a,
+            config: $STM_Config,
+            uid: ctx.session.uid,
+            serverBusy: false,
+            signup_bonus: '$1', // TODO: don't hardcode this
```

Comment:
Don't hardcode this...

Reply:
thx cpt. obvious :D 

@valzav / @roadscape do you know if `signup_bonus` is still used, would be nice if it can be removed

Reply:
The variable is still referenced and even passed to translator, but as far as I can tell the translations are no longer making use of it. The amount of STEEM rewarded to new users is low right now and I don't see it increasing since we rely on delegation now. Giving more directly on signup just encourages more abuse.